aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp

```diff
@@ -563,12 +605,20 @@ class QEmbedding final {
       const c10::intrusive_ptr<EmbeddingPackedParamsBase>& packed_weight,
       const Tensor& indices,
       bool pruned_weights) {
+    // Set default offsets here since the FBGEMM lookup op expects it.
     const auto offsets_size = indices.numel();
-    at::Tensor offsets = at::arange(0, offsets_size, at::kLong);
+    at::Tensor offsets = at::arange(0, offsets_size, indices.scalar_type());
```

Comment:
question: do we support other types of indices?

Reply:
You mean dtype? We support int32, int64 types